src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Query/PropertyPages/PropertyPageDataProducer.cs

```diff
@@ -99,16 +120,38 @@ public static IEntityValue CreatePropertyPageValue(IEntityRuntimeModel runtimeMo
 
             IEnumerable<IEntityValue> createPropertyPageValuesAsync()
             {
+                Rule? parentDebuggerPageRule = null;
+                
                 var propertyPageQueryContext = queryCacheProvider.CreateCache(project);
                 foreach (string schemaName in projectCatalog.GetProjectLevelPropertyPagesSchemas())
                 {
                     if (projectCatalog.GetSchema(schemaName) is Rule rule
                         && !rule.PropertyPagesHidden)
                     {
-                        IEntityValue propertyPageValue = CreatePropertyPageValue(parent, propertyPageQueryContext, rule, requestedProperties);
-                        yield return propertyPageValue;
+                        if (rule.PageTemplate == "debuggerParent")
+                        {
+                            parentDebuggerPageRule = rule;
+                        }
+                        else if (rule.PageTemplate == "commandNameBasedDebugger")
```

Comment:
nit: you can put this condition (inversed) in the outer if-statement so it is ignored upfront.